### PR TITLE
Update gradle wrapper, replace deprecated features in gradle build.

### PR DESCRIPTION
**Pull request by anurag-paliwal**
<h4> Update gradle wrapper, replace deprecated features in gradle build. </h4>
<br>
<h4> Description (Describe in detail the fix made) : </h4>

- Update gradle wrapper to 8.4
- Replace deprecated gradle features.
- Update README to make use of gradle wrapper.
- Update CI workflows to make use of bundled gradle wrapper.
- Added JDK resolver - which sets up compatible JDK in case system has older/incompatible Java installed.

<h4> Bug ID (if any) : b/303573290 </h4>
<br>
<h4> TESTED (Test Cases with scenario and description - must have 1 positive and 1 negative scenario) : </h4>
Tested common workflows on clean workspace with bundled wrapper.

### CODE REVIEW

File: gradlew [hunk 1]
```diff
@@ -83,10 +83,8 @@ done
 # This is normally unused
 # shellcheck disable=SC2034
 APP_BASE_NAME=${0##*/}
-APP_HOME=$( cd "${APP_HOME:-./}" && pwd -P ) || exit
-
-# Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
+# Discard cd standard output in case $CDPATH is set (https://github.com/gradle/gradle/issues/25036)
```
Comment (anurag-paliwal) [inline gradlew:86]: This was updated by gradle wrapper update command.

**Review by saurabhgadia4 — APPROVED** [review]

File: .github/workflows/dlp-pipelines.yml [hunk 2]
```diff
@@ -40,12 +40,13 @@ jobs:
       - uses: actions/checkout@v4
 
       - name: Setup Java
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
-          java-version: 11
+          distribution: 'zulu'
```
Comment (chitara-01) [inline .github/workflows/dlp-pipelines.yml:45]: Is there a reason to choose these specifications? We recommend using jdk 19.0.1 and above in README. Please make both the mentions compatible.

File: README.md [hunk 3]
```diff
@@ -126,7 +126,7 @@ projected usage.
 ### Compile the code
 
 ```
-gradle build
```
Comment (chitara-01) [inline README.md:129]: Should we add Catherine as a reviewer for the README changes?

**Review by chitara-01 — APPROVED** [review]
LGTM

File: .github/workflows/dlp-pipelines.yml (hunk 2 above)
Comment (anurag-paliwal) [reply .github/workflows/dlp-pipelines.yml:45]: Right now even with JDK 19+ - the code actually enforces use of compatibility mode and runs under JDK 17. To be able to use proper JDK 19+ support we'll have to make some code changes - which is why I've updated it to 17 for now - once we confirm 19 support we can update toolchain as well.

File: README.md [hunk 4]
```diff
@@ -649,12 +649,7 @@ This section provides a minimal list of tools and libraries along with their rec
 run this pipeline outside Google Shell.
 
 1. Install Java using [OpenJDK](https://openjdk.org/) (recommended version: 19.0.1 and above).
-1. Gradle - Use the bundled [gradle wrapper to install or update](https://docs.gradle.org/current/userguide/gradle_wrapper.html#sec:upgrading_wrapper) the required Gradle version automatically.
-    1. To update the Gradle version (recommended version: 8.3):
-    ```commandline
-    gradle wrapper --gradle-version <updated_version>
-    ```
-    2. Build with the updated version:
+1. Use Gradle wrapper to build the project:
```
Comment (catherinerosesantos) [inline README.md:652]: Gradle Wrapper

(They capitalize Wrapper in their docs so I think that's the proper name of the feature.)

**Review by catherinerosesantos — APPROVED** [review]

**Pull request merged**